File: codesprint/conn.py
n=input();
for i in range(0,n):
    k=input();
    a = [0] * k;
    b = [0] * k;
    j=0;
    #    
    a=raw_input().split();
    b=raw_input().split();
    t=k-1;
    r=0;
    while t>0 :
        y=t-1;
        while y>=0 :
              temp=abs(long(a[t])-long(a[y]));
              if long(b[t])>long(b[y]):
                r=r+temp*long(b[t]);
              else:
                r=r+temp*long(b[y]);
              
              y=y-1;
        t=t-1;
    print r%1000000007;
    
    
    #for str in raw_input().split():
    #    a[j]=long(str);
    #    j=j+1;
    #j=0;   
    #for str in raw_input().split():
    #    b[j]=long(str);
    #    j=j+1;

File: codesprint/conn.c
#include<stdio.h>
#include<math.h>
int main()
{
    long n=0,i=0,temp=0,j=0,y=0,c=0;
    long a[200000],b[200000];
    unsigned long long r=0;
    scanf("%ld",&n);
    for(i=0;i<n;i++)
    {   scanf("%ld",&c);
        for(j=0;j<c;j++)
        {
            scanf("%ld",&a[j]);
        }
        r=0;     
        for(j=0;j<c;j++)
        {
            scanf("%ld",&b[j]);
            y=0;
            while (y<j)
            {
              temp=abs(a[j]-a[y]);
              if (b[j]>b[y])
                r=(r+temp*b[j])%1000000007;
              else
                r=(r+temp*b[y])%1000000007;
              y=y+1;
             }
        }
        printf("\n%llu",r);   
   }
return 0;
}
